wandb/apis/public.py

```diff
@@ -2479,6 +2479,30 @@ def best_run(self, order=None):
         except IndexError:
             return None
 
+    def expected_run_count(self) -> int:
+        "Returns the number of expected runs in the sweep or -1 for infinite runs."
+        search_method: str = self.config.get("method", None)
+        if not search_method == "grid":
+            wandb.termwarn(
+                f"Sweep of method {search_method} does not have a constrained expected runs."
+            )
+            return -1
+
+        def _recursive_count(params: Dict[str, Any], count: int) -> int:
+            for param_dict in params.get("parameters", {}).values():
+                if param_dict.get("parameters", None) is not None:
+                    count *= _recursive_count(param_dict["parameters"], count)
+                elif param_dict.get("value", None) is not None:
+                    count *= 1
+                elif param_dict.get("values", None) is not None:
+                    count *= len(param_dict["values"])
+                else:
+                    # Even one non discrete param will result in infinite expected runs
+                    return -1
+            return count
+
+        return _recursive_count(self.config, 1)
+
```

Comment:
we should just call the graphql API endpoint that you added in the other PR.

Reply:
Since this is already implemented w/ tests could we just use this solution? This was a low prio request to begin with so rewriting it doesn't seem like good value for time. There might even be an argument for having something that can run locally in python without having to send a request over the internet.

Reply:
its a two liner - and we should ensure that our APIs are consistent